emailCrawler.java

```diff
@@ -31,14 +24,13 @@
 
 	public static void main(String args[]) throws IOException {
 		String url = "http://" + args[0];
-		// this list is used to store all of urls which can be discovered in
-		// home page. Actually, I can write this program
-		// by using DFS so that I can find more discoverable pages on discovered
-		// pages. But I don't do it for that it may go into a infinite loop.
+		// this list is used to store all of urls which can be discovered in home page. Actually, I can write this 
+		// program by using DFS so that I can find more discoverable pages on pages which we have discovered. 
+		// But here, I don't do it for that it may go into a infinite loop.
 		ArrayList<String> urlList = new ArrayList<>();
```

Comment:
Because I just get links from home page so I don't think about duplicate URLs. However, if we want to avoid duplicates, we can use a HashSet to store URLs which we have visited and if we found a new URL which is contained in this HashSet, we can pass it. That's a way to avoid duplicate URLs.